### PR TITLE
Enable dask dispatch to cuDF's `is_categorical_dtype` for cuDF objects

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Fixes https://github.com/rapidsai/cudf/issues/7111
Closes https://github.com/rapidsai/cudf/issues/7400

### CODE REVIEW

**Review by kkraus14 — APPROVED** [review]
Can we add a test in the dask-cudf unit tests for the `get_dummies` method which tests this?

**Comment by kkraus14** [comment]
We'll also need to bump the required Dask version here

**Review by galipremsagar — APPROVED** [review]

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
Tests look good. Will push a commit bumping the version in the conda recipe and this should be good to go.

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -83,8 +83,8 @@ gpuci_conda_retry install -y \
                   "ucx-py=${MINOR_VERSION}"
 
 # https://docs.rapids.ai/maintainers/depmgmt/
-# gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
-# gpuci_conda_retry install -y "your-pkg=1.0.0"
+gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
+gpuci_conda_retry install -y "dask>=2021.3.1"
```
Comment (kkraus14) [inline ci/gpu/build.sh:87]: We already install dask / distributed from the `main` branch from github below so this is not needed

**Comment by brandon-b-miller** [comment]
created https://github.com/rapidsai/integration/pull/235

File: ci/gpu/build.sh (hunk 1 above)
Comment (kkraus14) [reply ci/gpu/build.sh:87]:
```suggestion
# gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
# gpuci_conda_retry install -y "your-pkg=1.0.0"
```

**Review by raydouglass — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**